utils/render-rule.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python3
+
+from glob import glob
+import collections
+import sys
+import pathlib
+
+import pygments
+import pygments.lexers
+import pygments.formatters
+
+import ssg.build_yaml
+import ssg.controls
+import ssg.yaml
+import ssg.jinja
+import template_renderer
+from rendering import common
+
+
+POLICY_FIELDS_MAPPING = dict(
+    checktext="checks",
+    fixtext="fixes",
+    srg_requirement="prose",
+    vuldiscussion="prose",
+)
+
+
+class Namespace:
+    def __init__(self):
+        self.policy_specific_content = collections.defaultdict(dict)
+
+    def contains_something(self):
+        copy_of_attributes = dict(vars(self))
+
+        if self.policy_specific_content:
+            return True
+        else:
+            copy_of_attributes.pop("policy_specific_content")
+
+        for stuff in copy_of_attributes:
+            if stuff:
+                return True
+        return False
+
+
+class HtmlOutput(template_renderer.Renderer):
+    TEMPLATE_NAME = "rendering/rule-template.html"
+
+    def process_rule(self, rule_id):
+        rule_fname = pathlib.Path(self.built_content_path) / "rules" / (rule_id + ".yml")
+        try:
+            rule = ssg.build_yaml.Rule.from_yaml(rule_fname, self.env_yaml)
+        except RuntimeError as exc:
+            msg = (
+                f"{str(exc)}. Make sure that the product is compiled, "
+                f"and '{rule_id}' is a short rule ID.")
+            print(msg, file=sys.stderr)
+            sys.exit(1)
+
+        prose = Namespace()
+        prose.description = common.resolve_var_substitutions(rule.description)
+        prose.rationale = common.resolve_var_substitutions(rule.rationale)
+
+        fixes = Namespace()
+        built_content_path = pathlib.Path(self.built_content_path)
+        bash_loc = built_content_path / "fixes" / "bash" / (rule_id + ".sh")
+        fixes.bash = self._highlight_file(bash_loc, pygments.lexers.BashLexer())
+        ansible_loc = built_content_path / "fixes" / "ansible" / (rule_id + ".yml")
+        fixes.ansible = self._highlight_file(ansible_loc, pygments.lexers.YamlLexer())
+
+        checks = Namespace()
+        oval_loc = built_content_path / "checks" / "oval" / (rule_id + ".xml")
```

Comment:
Also keep in mind that some rules are templated so they use a different directory.